crates/electrum/tests/test_electrum.rs

```diff
@@ -272,32 +292,91 @@ fn scan_detects_confirmed_tx() -> anyhow::Result<()> {
     // Mine some blocks.
     env.mine_blocks(101, Some(addr_to_mine))?;
 
-    // Create transaction that is tracked by our receiver.
+    // Broadcast transaction to mempool.
     env.send(&addr_to_track, SEND_AMOUNT)?;
+    env.wait_until_electrum_sees_block()?;
+
+    electrum_sync(
+        &client,
+        &[spk_to_track.clone()],
+        5,
+        &mut recv_chain,
+        &mut recv_graph,
+    )?;
+
+    // Check if balance is zero when transaction exists only in mempool.
+    assert_eq!(
+        get_balance(&recv_chain, &recv_graph)?,
+        Balance {
+            confirmed: Amount::from_sat(0),
+            ..Balance::default()
+        },
+        "initial balance must be correct",
+    );
 
-    // Mine a block to confirm sent tx.
+    // Mine block to confirm transaction.
     env.mine_blocks(1, None)?;
+    env.wait_until_electrum_sees_block()?;
 
-    // Sync up to tip.
+    electrum_sync(
+        &client,
+        &[spk_to_track.clone()],
+        5,
+        &mut recv_chain,
+        &mut recv_graph,
+    )?;
+
+    // Check if balance is correct when transaction is confirmed.
+    assert_eq!(
+        get_balance(&recv_chain, &recv_graph)?,
+        Balance {
+            confirmed: SEND_AMOUNT,
+            ..Balance::default()
+        },
+        "initial balance must be correct",
+    );
+
+    // Perform reorg on block with confirmed transaction.
+    env.reorg_empty_blocks(1)?;
     env.wait_until_electrum_sees_block()?;
-    let update = client.sync(
-        SyncRequest::from_chain_tip(recv_chain.tip()).chain_spks(core::iter::once(spk_to_track)),
+
+    electrum_sync(
+        &client,
+        &[spk_to_track.clone()],
         5,
-        true,
+        &mut recv_chain,
+        &mut recv_graph,
     )?;
 
-    let _ = recv_chain
-        .apply_update(update.chain_update)
-        .map_err(|err| anyhow::anyhow!("LocalChain update error: {:?}", err))?;
-    let _ = recv_graph.apply_update(update.graph_update);
+    // Check if balance is correct when transaction returns to mempool.
+    assert_eq!(
+        get_balance(&recv_chain, &recv_graph)?,
+        Balance {
+            confirmed: Amount::from_sat(0),
+            ..Balance::default()
+        },
+    );
+
+    // Mine block to confirm transaction again.
+    env.mine_blocks(1, None)?;
+    env.wait_until_electrum_sees_block()?;
+
+    electrum_sync(
+        &client,
+        &[spk_to_track],
+        5,
+        &mut recv_chain,
+        &mut recv_graph,
+    )?;
 
-    // Check to see if tx is confirmed.
+    // Check if balance is correct once transaction is confirmed again.
     assert_eq!(
         get_balance(&recv_chain, &recv_graph)?,
         Balance {
             confirmed: SEND_AMOUNT,
             ..Balance::default()
         },
+        "initial balance must be correct",
     );
 
     for tx in recv_graph.graph().full_txs() {
```

Comment:
We need a comment before the for loop to explain what is being tested in the for loop.